### PR TITLE
修改pymongo认证失败问题

### LINKED ISSUE #2641: 无法指定Mongo实例操作：`'Collection' object is not callable. If you meant to call the 'authenticate' method on a 'Database' object it is failing because no such method exists.`
### 重现步骤

- 会话管理(进程状态)/在线查询/Sql上线 等等
- 选择mongo实例

### 预期外的结果

image

### 日志文本

```shell
2024-05-14 10:19:35,920 - default - WARNING - mongodb获取连接信息错误，错误信息Traceback (most recent call last):
  File "/opt/archery/sql/engines/mongo.py", line 1206, in current_op
    conn = self.get_connection()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/archery/sql/engines/mongo.py", line 806, in get_connection
    self.conn[self.db_name].authenticate(self.user, self.password, auth_db)
  File "/opt/venv4archery/lib/python3.11/site-packages/pymongo/collection.py", line 3501, in __call__
    raise TypeError(
TypeError: 'Collection' object is not callable. If you meant to call the 'authenticate' method on a 'Database' object it is failing because no such method exists
```
```


### 版本

1.11.1

### 部署方式

Docker

### 是否还有其他可以辅助定位问题的信息？比如数据库版本等

_No response_

### PR DESCRIPTION
**Pull request by woshiyanghai**
fix https://github.com/hhyo/Archery/issues/2641

### CODE REVIEW

File: sql/engines/mongo.py [hunk 1]
```diff
@@ -795,15 +795,22 @@ def execute_check(self, db_name=None, sql=""):
...
-        )
+        # self.conn = pymongo.MongoClient(
+        #     self.host,
+        #     self.port,
+        #     authSource=auth_db,
+        #     connect=True,
+        #     connectTimeoutMS=10000,
+        # )
+        # if self.user and self.password:
+        #     self.conn[self.db_name].authenticate(self.user, self.password, auth_db)
```
Comment (LeoQuote) [inline sql/engines/mongo.py]: 请直接删除即可, 无需注释

**Comment by sunbluers** [comment]
删除后提示如下错误：
{status: 1, msg: "无法连接实例,↵'NoneType' object has no attribute 'list_database_names'", data: []}
data
: 
[]
msg
: 
"无法连接实例,\n'NoneType' object has no attribute 'list_database_names'"
status
: 
1

**Comment by sunbluers** [comment]
返回错误信息：
{status: 1,…}
data
: 
[]
msg
: 
"无法连接实例,\nReserved characters such as ':' must be escaped according RFC 2396. An IPv6 address literal must be enclosed in '[' and ']' according to RFC 2732."
status
: 
1

**Comment by sunbluers** [comment]
由于mongo版本较低，需要pymongo版本适配，最终fix.

**Comment by wang-xiaowu** [comment]
该pr引起了另一个问题：
image

见：https://github.com/hhyo/Archery/pull/2645

**Comment by woshiyanghai** [comment]
让另一个朋友pr这里吧

**Pull request closed without merging**